### PR TITLE
Fuzzing: Improve coverage

### PR DESCRIPTION
**Pull request by AdamKorcz**
**Please provide a description of this PR:**
Adds more fuzzers to improve test coverage.
From a high level, more coverage is being added for the following areas of the code:

- Kube CRD
- Aggregate controller
- Kube gateway controller
- Status reconciler
- Workload entry controller
- CA Server
- Security utilities
- Security pki utilities
- Pilot status

This PR does not modify any production code. It only adds new fuzz tests.

The PR does remove the analyzer fuzzer from the regression tests since it broke it because of the recent major refactoring. I will take it upon myself to add it back but would prefer to do this in a separate PR, since I have some things to add to this fuzzer as well.

### CODE REVIEW

**Comment by AdamKorcz** [comment]
Could we move forward with this PR as well?

File: tests/fuzz/analyzer_fuzzer.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func runAnalyzer(sa *local.IstiodAnalyzer) (local.AnalysisResult, error) {
 }
 
 // FuzzAnalyzer implements the fuzzer
-func FuzzAnalyzer(data []byte) int {
+func FFuzzAnalyzer(data []byte) int {
```
Comment (howardjohn) [inline tests/fuzz/analyzer_fuzzer.go:96]: Why is this changed?

**Review by howardjohn — COMMENTED** [review]
lgtm except the FuzzAnalyzer part I am not sure of

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
Missed the comment

> The PR does remove the analyzer fuzzer from the regression tests since it broke it because of the recent major refactoring. I will take it upon myself to add it back but would prefer to do this in a separate PR, since I have some things to add to this fuzzer as well.

Is this just a merge conflict maybe? The expected state currently is the fuzzer is enabled, and has no 'brokenCases' as we fixed those in the main codebase

**Comment by AdamKorcz** [comment]
> Is this just a merge conflict maybe?

No, it is related to the recent refactoring. If you would rather not have a fuzzer that _should_ be tested but _is not_ being tested, then I am fine with removing/deactivating it completely and add it back later.
Fixing this fuzzer will fit nicer with adding the updated analyzer fuzzer I had initially uploaded in this PR but which now needs some adjustments.

File: tests/fuzz/analyzer_fuzzer.go (hunk 1 above)
Comment (AdamKorcz) [reply tests/fuzz/analyzer_fuzzer.go:96]: Marking as solved because of [this comment](https://github.com/istio/istio/pull/35900#issuecomment-971786345).

**Comment by howardjohn** [comment]
> No, it is related to the recent refactoring. There were also merge conflicts, but it seemed to be broken in several ways. If you would rather not have a fuzzer that should be tested but is not being tested, then I am fine with removing/deactivating it completely and add it back later.

How is it broken or not tested?

All of the fuzzers are run in the oss-fuzz, and all fuzzers are tested in unit tests with a set of simple smoke-test cases. In addition, tests that have ever failed are run against those cases. Some also have *known broken cases* (ie bugs in our code). The recent PR to TestAnalyzer *fixed* a bug, causing the known broken cases to be reduced. The test coverage is still the same

**Comment by AdamKorcz** [comment]
> How is it broken or not tested?

I meant that this PR will result in the analyzer fuzzer not being included in the regression tests.

**Comment by AdamKorcz** [comment]
Adding the analyzer fuzzer back to the regression test.

File: tests/fuzz/pki_fuzzer.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+func FuzzParsePemEncodedCertificateChain(data []byte) int {
+	_, _ = util.ParsePemEncodedCertificateChain(data)
+	return 1
+}
+
+// FuzzUpdateVerifiedKeyCertBundleFromFile implements a
+// fuzzer that tests UpdateVerifiedKeyCertBundleFromFile
+func FuzzUpdateVerifiedKeyCertBundleFromFile(data []byte) int {
+	f := fuzz.NewConsumer(data)
+	certFile, err := os.Create("certfile")
```
Comment (howardjohn) [inline tests/fuzz/pki_fuzzer.go:100]: can we delete it on cleanup?

Also is using files for fuzzing really appropriate? Seems like we should just fuzz NewVerifiedKeyCertBundleFromPem instead?

File: tests/fuzz/regression_test.go [hunk 3]
```diff
@@ -35,7 +35,10 @@ var baseCases = [][]byte{
 
 // brokenCases contains test cases that are currently failing. These should only be added if the
 // failure is publicly disclosed!
-var brokenCases = map[string]string{}
+var brokenCases = map[string]string{
+	"6169070276837376": "https://github.com/go-yaml/yaml/issues/666",
+	"6087702507290624": "https://github.com/go-yaml/yaml/issues/768",
+}
```
Comment (howardjohn) [inline tests/fuzz/regression_test.go:41]: If you revert this change I think things will get into the expected state.

Comment (AdamKorcz) [reply tests/fuzz/regression_test.go]: thanks!

File: tests/fuzz/pki_fuzzer.go (hunk 2 above)
Comment (AdamKorcz) [reply tests/fuzz/pki_fuzzer.go:100]: > can we delete it on cleanup?

Sure, added.

> Also is using files for fuzzing really appropriate?

Yes, it is fine as long as it is fast enough. I suggest we try `UpdateVerifiedKeyCertBundleFromFile` and observe if the speed is poor after a few weeks. Otherwise we stick with `NewVerifiedKeyCertBundleFromPem`.
